validator_client/src/validator_directory.rs

```diff
@@ -0,0 +1,404 @@
+        if !directory.exists() {
+            return Err(format!(
+                "Validator directory does not exist: {:?}",
+                directory
+            ));
+        }
+
+        Ok(Self {
+            voting_keypair: Some(
+                load_keypair(directory.clone(), VOTING_KEY_PREFIX)
+                    .map_err(|e| format!("Unable to get voting keypair: {}", e))?,
+            ),
+            withdrawal_keypair: load_keypair(directory.clone(), WITHDRAWAL_KEY_PREFIX).ok(),
+            deposit_data: load_eth1_deposit_data(directory.clone()).ok(),
+            directory,
+        })
+    }
+}
+
+/// Load a `Keypair` from a file.
+fn load_keypair(base_path: PathBuf, file_prefix: &str) -> Result<Keypair, String> {
+    let path = base_path.join(keypair_file(file_prefix));
+
+    if !path.exists() {
+        return Err(format!("Keypair file does not exist: {:?}", path));
+    }
+
+    let mut bytes = vec![];
+
+    File::open(&path)
+        .map_err(|e| format!("Unable to open keypair file: {}", e))?
+        .read_to_end(&mut bytes)
+        .map_err(|e| format!("Unable to read keypair file: {}", e))?;
+
+    SszEncodableKeypair::from_ssz_bytes(&bytes)
+        .map(Into::into)
+        .map_err(|e| format!("Unable to decode keypair: {:?}", e))
+}
+
+/// Load eth1_deposit_data from file.
+fn load_eth1_deposit_data(base_path: PathBuf) -> Result<Vec<u8>, String> {
+    let path = base_path.join(ETH1_DEPOSIT_DATA_FILE);
+
+    if !path.exists() {
+        return Err(format!("Eth1 deposit data file does not exist: {:?}", path));
+    }
+
+    let mut bytes = vec![];
+
+    File::open(&path)
+        .map_err(|e| format!("Unable to open eth1 deposit data file: {}", e))?
+        .read_to_end(&mut bytes)
+        .map_err(|e| format!("Unable to read eth1 deposit data file: {}", e))?;
+
+    let string = String::from_utf8_lossy(&bytes);
+    if string.starts_with("0x") {
+        hex::decode(&string[2..])
+            .map_err(|e| format!("Unable to decode eth1 data file as hex: {}", e))
+    } else {
+        Err(format!("String did not start with 0x: {}", string))
+    }
+}
+
+/// A helper struct to allow SSZ enc/dec for a `Keypair`.
+#[derive(Encode, Decode)]
+struct SszEncodableKeypair {
+    pk: PublicKey,
+    sk: SecretKey,
+}
+
+impl Into<Keypair> for SszEncodableKeypair {
+    fn into(self) -> Keypair {
+        Keypair {
+            sk: self.sk,
+            pk: self.pk,
+        }
+    }
+}
+
+impl From<Keypair> for SszEncodableKeypair {
+    fn from(kp: Keypair) -> Self {
+        Self {
+            sk: kp.sk,
+            pk: kp.pk,
+        }
+    }
+}
+
+/// Builds a `ValidatorDirectory`, both in-memory and on-disk.
+#[derive(Default)]
+pub struct ValidatorDirectoryBuilder {
+    directory: Option<PathBuf>,
+    voting_keypair: Option<Keypair>,
+    withdrawal_keypair: Option<Keypair>,
+    amount: Option<u64>,
+    deposit_data: Option<Vec<u8>>,
+    spec: Option<ChainSpec>,
+}
+
+impl ValidatorDirectoryBuilder {
```

Comment:
Add more comments to these fns

Reply:
I have done this over in another branch that should get merged into Scott's slashing PR:

https://github.com/sigp/lighthouse/tree/paul-slash-protection

I won't add them again here cause it'll cause conflicts.